skimage/io/_plugins/matplotlib_plugin.py

```diff
@@ -169,10 +172,29 @@ def imshow_collection(ic, *args, **kwargs):
     """Display all images in the collection.
 
     """
-    fig, axes = plt.subplots(1, len(ic))
+    # N : total images.
+    # Aim : 4 nrows = 3 ncols
+    # 4r = 3c = k
+    # rc = N, => k = (12 N)^(0.5)
+    # r = floor(k/4) or floor(k/4) + 1
+    # c = N/r or N/r + 1
+    # Choose the pair which is closer to 3:4 = 0.75
+    N = len(ic)
+    k = (N * 12)**(0.5)
+    r1 = floor(k/4)
+    r2 = r1 + 1
+    c1 = ceil(N/r1)
+    c2 = ceil(N/r2)
+    if abs(r1/c1 - 0.75) < abs(r2/c2 - 0.75):
+        nrows, ncols = r1, c1
+    else:
+        nrows, ncols = r2, c2
+    fig = plt.figure()
```

Comment:
`plt.subplots` is a better option here.  You may also want to use the `sharex` and `sharey` kwargs if the images are all the same size (sorry, parachuting in with out _really_ knowing what is going on here :smiling_imp: )